### ISSUE TITLE
Add support for plotting 2D decision regions for a slice in feature space 

### ISSUE THREAD

**Issue by jrbourbeau** [rasbt/mlxtend#188]
I really like the `plot_decision_regions` functionality of mlxtend. However, the current version only supports up to two training features. I often find myself wanting to plot a 2D decision region for a slice in a higher-dimensional (>2) feature space.

As a quick example, I used `sklearn.datasets.make_blobs` (with scikit-learn version 0.18.1) to generate a dataset with 3 training features

```
X, y = make_blobs(n_samples=10000, n_features=3, centers=[[2, 2, -2],[-2, -2, 2]],
                  cluster_std=[2, 2], random_state=2)
```

Below are several 2D decision regions (using a grid for the first two features) for constant values of the third feature. 

![image](https://cloud.githubusercontent.com/assets/11656932/25924207/2a7064a4-35a7-11e7-9720-ce0b72f24580.png)

If you think this is a feature worth adding, I can open up a pull request with the code used to make the above plots. Thanks!

**Comment by rasbt** [comment]
I like that idea! However, I think that we shouldn't add this feature via an additional parameter to the `plot_decision_regions` function as it is already be quite complex. What do you think about a function called `plot_decision_region_slices` that wraps the `plot_decision_regions` function to produce a grid of the different 2D feature combinations?

E.g.,

```python
plot_decision_region_slices(..., combinations=None)
```
would produce a `m x 2` (or `m x 3`) grid of all possible feature combinations, e.g., via sth like 

```python
>>> feature_ary.shape
(100, 4)
>>> from itertools import combinations
>>> list(combinations(np.arange(feature_ary.shape[1]), r=2))
[(0, 1), (0, 2), (0, 3), (1, 2), (1, 3), (2, 3)]
```

Alternatively, a user could provide specific slices, e.g,. 

```python
plot_decision_region_slices(..., combinations=[(0, 1), (1, 3), (2, 3)])
```

What do you think?

**Comment by jrbourbeau** [comment]
I totally agree that there should be a separate `plot_decision_region_slices` function. Unfortunately there is the added complication that you need to specify filler values for the other features that aren't on the x-y axis. I'll submit a pull request with the current incarnation of the `plot_decision_region_slices` code I have.

**Comment by rasbt** [comment]
> there is the added complication that you need to specify filler values

Oh yeah, that's because of the scikit API, right?  Let me write further comments to the PR :)

**Comment by rasbt** [comment]
Adressed in #189
